src/meta/service/src/ddl_service.rs

```diff
@@ -747,64 +738,11 @@ impl DdlService for DdlServiceImpl {
             return Err(Status::invalid_argument("request is empty"));
         }
 
-        match req.payload.unwrap() {
-            create_connection_request::Payload::PrivateLink(link) => {
-                // currently we only support AWS
-                let private_link_svc = match link.get_provider()? {
-                    PbPrivateLinkProvider::Mock => PbPrivateLinkService {
-                        provider: link.provider,
-                        service_name: String::new(),
-                        endpoint_id: String::new(),
-                        endpoint_dns_name: String::new(),
-                        dns_entries: HashMap::new(),
-                    },
-                    PbPrivateLinkProvider::Aws => {
-                        if let Some(aws_cli) = self.aws_client.as_ref() {
-                            let tags_env = self
-                                .env
-                                .opts
-                                .privatelink_endpoint_default_tags
-                                .as_ref()
-                                .map(|tags| {
-                                    tags.iter()
-                                        .map(|(key, val)| (key.as_str(), val.as_str()))
-                                        .collect()
-                                });
-                            aws_cli
-                                .create_aws_private_link(
-                                    &link.service_name,
-                                    link.tags.as_deref(),
-                                    tags_env,
-                                )
-                                .await?
-                        } else {
-                            return Err(Status::from(MetaError::unavailable(
-                                "AWS client is not configured",
-                            )));
-                        }
-                    }
-                    PbPrivateLinkProvider::Unspecified => {
-                        return Err(Status::invalid_argument("Privatelink provider unspecified"));
-                    }
-                };
-                let connection = Connection {
-                    id: 0,
-                    schema_id: req.schema_id,
-                    database_id: req.database_id,
-                    name: req.name,
-                    owner: req.owner_id,
-                    info: Some(connection::Info::PrivateLinkService(private_link_svc)),
-                };
-
-                // save private link info to catalog
-                let version = self
-                    .ddl_controller
-                    .run_command(DdlCommand::CreateConnection(connection))
-                    .await?;
-
-                Ok(Response::new(CreateConnectionResponse { version }))
-            }
-        }
+        return match req.payload.unwrap() {
```

Comment:
A `create connection` of PrivateLink should already be rejected in `resolve_create_connection_payload`. Please `panic` or `assert` here (as well as other occurances).